### DISCUSSION TITLE
Master-Detail-Pages

### DISCUSSION THREAD

**Discussion by SaschaValentin** [mavoweb/mavo#948] [Q&A] [answered]
Dear community,

I'm looking for a way to have something like a Master-Detail-Connection with Mavo. I mean something like a blog where a link from one of the posts on the listing/overview page leads to the detail post. The data for the posts should come from a collection.

Can something like this achieved with Mavo?

Hope, someone can help me and give me a little hint of this is possible.

Thanks a lot

Sascha

**Comment by DmitrySharabin** [comment, marked as answer]
Hey Sascha,

I have a kinda work-in-progress-project from far 2020 that covers the exact cases you need (if I get everything correctly). I'll be happy if you find it useful.

Here is the repo: https://github.com/DmitrySharabin/mavo-powered-blog
And here is the site deployed on Netlify: https://mavo-powered-blog.netlify.app/

Feel free to ask any questions you might have!

**Comment by DmitrySharabin** [comment]
And one more thing—to login, please add `?login` at the end of the site URL.

Good luck! 😊

**Comment by SaschaValentin** [comment]
Hey Dmitry,

thanks for your fast answer. Yes, this seems to be exactly what I was looking for. I will study it a bit and maybe contact you again if I have more questions.

Have a nice week

**Comment by DmitrySharabin** [comment]
@SaschaValentin I found one more demo that also might help you: https://github.com/DmitrySharabin/mavo-multi-page-technique

**Comment by SaschaValentin** [comment]
Dear Dmitry,

once again thanks for your answer. Over the weekend I found time to try it out and it works great. So much thanks.

But there is another question and maybe you can help me to understand the concept (but maybe that doesn´t work at all - all I need is a little hint :-)). I wonder if it is possible to have real-time updating the data on the page. Data is provided by Appwrite or RethinkDB, both are capable of doing real-time communication. But I can´t imagine how to include this on an page and if this works with Mavo. So I thought maybe you are a bit more expert on this ;-) or just can tell me, if this is even possible.

Best wishes

Sascha

**Comment by DmitrySharabin** [comment]
Hey Sascha,

I'm happy to know that it worked. Yay! 🎉

> I wonder if it is possible to have real-time updating the data on the page.

Long story short, it is possible. Actually, the [Mavo Chat](https://dmitrysharabin.github.io/mavo-chat/) app is built based on this feature. Its [source code](https://github.com/DmitrySharabin/mavo-chat) is on GitHub.

However, we can use this feature only with backends that support it. E.g., Mavo Chat uses [the Firebase Firestore backend](https://plugins.mavo.io/plugin/firebase-firestore) for this.

Unfortunately, Mavo doesn't support any backend services you mentioned: `Appwrite` and `RethinkDB`. Please feel free to [submit PRs to request corresponding plugins](https://github.com/mavoweb/plugins/issues). Who knows, maybe someday we'll get these services supported. 😊

Meanwhile, there is a workaround that might help, but it involves custom JavaScript. If, by any chance, you know how to listen to real-time updates on your data using the API provided by those services, you may fire the `mv-remotedatachange` event every time you get new data and pass that new data as a value of the `data` property of the event.

For example, every document on Firestore has the `onSnapshot` method that allows you to [listen to changes](https://firebase.google.com/docs/firestore/query-data/listen) on the corresponding document. So by subscribing to those changes, we can get real-time updates.

I built [a small app](https://codepen.io/dmitrysharabin/pen/xxyYNxE) to illustrate the usage of the `mv-remotedatachange` event. It simulates the case when new data arrives and shows a way to tell Mavo about it. It might help to get the main idea.

Hope it helps!

In any case, feel free to ask any questions that might arise.

Good luck!

**Comment by SaschaValentin** [comment]
Dear Dmitry,

thanks again for your reply and help in this case.

I followed your advice and started a request for this. Now it´s time to hope someone is also interested in such a thing.

Until then I´ll take a look a CouchDB. On the Mavo plugin repository it is said, CouchDB is also capable of real-time data syncing. Maybe that could be a solution. I´ll give it a try.

Again: Thanks for your help.
